src/Mod/Arch/ArchRoof.py

```diff
@@ -286,393 +383,455 @@ def makeRoofProfilsDic(self, id, angle, run, idrel, overhang, thickness,):
-        ptInter = edgeHip.Vertexes[0].Point
-        vecInterRidges = DraftGeomUtils.findPerpendicular(ptInter, [profilCurrent["ridge"].Edges[0],], force=0)
-        ptInterRidges = ptInter.add(vecInterRidges[0])
-        self.ptsPaneProject.append(FreeCAD.Vector(ptInterRidges))
-
-    def backSmaller(self, i):
-
-        import DraftGeomUtils
-        profilCurrent = self.findProfil(i)
-        profilBack1 = self.findProfil(i-1)
-        dec = profilBack1["height"]/math.tan(math.radians(profilCurrent["angle"]))
-        edgeRidgeOnPane = DraftGeomUtils.offset(profilCurrent["edge"],self.getPerpendicular(profilCurrent["vec"],profilCurrent["rot"],dec))
-        ptInter1 = DraftGeomUtils.findIntersection(edgeRidgeOnPane,profilBack1["ridge"],infinite1=True,infinite2=True,)
-        edgeHip = DraftGeomUtils.edg(FreeCAD.Vector(ptInter1[0]),profilCurrent["edge"].Vertexes[0].Point)
-        ptInter2 = edgeHip.Vertexes[0].Point
-        vecInterRidges = DraftGeomUtils.findPerpendicular(ptInter2, [profilCurrent["ridge"].Edges[0],], force=0)
-        ptInterRidges = ptInter2.add(vecInterRidges[0])
-        self.ptsPaneProject.append(FreeCAD.Vector(ptInterRidges))
-        self.ptsPaneProject.append(FreeCAD.Vector(ptInter1[0]))
-        ptInterHipEave = DraftGeomUtils.findIntersection(edgeHip,profilCurrent["eave"],infinite1=True,infinite2=False,)
-        if ptInterHipEave:
-            self.ptsPaneProject.append(FreeCAD.Vector(ptInterHipEave[0]))
-        else:
-            ptInterHipEave = DraftGeomUtils.findIntersection(edgeHip,profilBack1["eaveD"],infinite1=True,infinite2=True,)
-            self.ptsPaneProject.append(FreeCAD.Vector(ptInterHipEave[0]))
-            self.ptsPaneProject.append(FreeCAD.Vector(profilCurrent["eave"].Vertexes[0].Point[0]))
+    def nextHigher(self, i):
+        profilCurr = self.findProfil(i)
+        profilNext = self.findProfil(i + 1)
+        dec = profilCurr["height"] / math.tan(math.radians(profilNext["angle"]))
+        per = self.getPerpendicular(profilNext["vec"], profilNext["rot"], dec)
+        edgeRidgeOnPane = DraftGeomUtils.offset(profilNext["edge"], per)
+        self.helperSloped(profilCurr,
+                          profilNext,
+                          profilCurr["ridge"],
+                          edgeRidgeOnPane,
+                          isBack = False)
+
+    def backLower(self, i):
+        profilCurr = self.findProfil(i)
+        profilBack = self.findProfil(i - 1)
+        dec = profilBack["height"] / math.tan(math.radians(profilCurr["angle"]))
+        per = self.getPerpendicular(profilCurr["vec"], profilCurr["rot"], dec)
+        edgeRidgeOnPane = DraftGeomUtils.offset(profilCurr["edge"], per)
+        self.helperSloped(profilCurr,
+                          profilBack,
+                          edgeRidgeOnPane,
+                          profilBack["ridge"],
+                          isBack = True,
+                          otherIsLower = True)
```

Comment:
Same situation here, but now with the first element `edgeRidgeOnPane` now based on `profilCurr`.

Reply:
OK, no need to repeat yourself to this extent.